File: Day3/Puzzel1.py
priority = 0;
input = []

with open("input.txt") as inputFile:
    for line in inputFile:
        input.append(line.strip())


for line in input:
    alreadyFound=""
    for i in range(0, int(len(line)/2)):
        currentChar = line[i]
        for j in range(int(len(line)/2), len(line)):
            if currentChar == line[j]:
                if currentChar not in alreadyFound:
                    alreadyFound += currentChar
                    if currentChar >= 'a' and currentChar <= 'z':
                        priority += ord(currentChar)-ord('a')+1        
                    elif currentChar >= 'A' and currentChar <= 'Z':
                        priority += ord(currentChar)-ord('A')+26+1            
                    else:
                        print( f"unexpected character found: {currentChar}\n")

print( f"Priority = {priority}" )

File: 2022/Day3/Puzzel1.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>

int main()
{
    std::ifstream inputFile("input.txt");
    std::vector<std::string> input = {};
    std::string temp;
    int priority = 0;
    while( std::getline(inputFile, temp))
    {
        input.push_back(temp);
    }
    for(auto line: input )
    {
        std::string alreadyFound="";
        for( int i=0; i<line.size()/2; i++)
        {
            char currentChar = line[i];
            for(unsigned int x=line.size()/2; x < line.size(); x++)
            {
                if( currentChar == line[x])
                {
                    bool found=false;
                    for( unsigned int y=0; y<alreadyFound.size()&&!found; y++)
                    {
                        if( currentChar==alreadyFound[y] )
                        {
                            found = true;
                            break;
                        }
                    }
                    if( !found )
                    {
                        alreadyFound += currentChar;
                        if (currentChar >= 'a' && currentChar <= 'z')
                            priority+=currentChar-'a'+1;
                        else if(currentChar >= 'A' && currentChar <= 'Z')
                            priority+=currentChar-'A'+26+1;
                        else
                            std::cerr<<"unexpected character found: " << currentChar << '\n';
                    }
                }   
            }
        }
    }
    std::cout << "Priority = " << priority << '\n';
}